src/main/scala/com/fulcrumgenomics/umi/DuplexConsensusCaller.scala

```diff
@@ -117,23 +117,55 @@ class DuplexConsensusCaller(override val readNamePrefix: String,
       errorRatePreUmi         = this.errorRatePreUmi,
       errorRatePostUmi        = this.errorRatePostUmi,
       minReads                = 1,
+      maxReads                = maxReads,
       minInputBaseQuality     = this.minInputBaseQuality,
       minConsensusBaseQuality = PhredScore.MinValue,
       producePerBaseTags      = true
     ))
 
   /**
-    * Returns the MI tag minus the trailing suffix that identifies /A vs /B
+    * Returns the MI tag **with** the trailing suffix that identifies /A vs /B
     */
-  override protected[umi] def sourceMoleculeId(rec: SamRecord): String = {
-    rec.get[String](ConsensusTags.MolecularId) match {
-      case None =>
+  private def sourceMoleculeAndStrandId(rec: SamRecord): String = {
+    // Optimization: speed up retrieving this tag by storing it in the transient attributes
+    rec.transientAttrs.getOrElse[String](ConsensusTags.MolecularId, {
+      val mi = rec.getOrElse[String](ConsensusTags.MolecularId,
         throw new IllegalStateException(s"Read ${rec.name} is missing it's ${ConsensusTags.MolecularId} tag.")
-      case Some(mi) =>
-        val index = mi.lastIndexOf('/')
-        require(index > 0, s"Read ${rec.name}'s $ConsensusTags tag doesn't look like a duplex id: $mi")
-        mi.substring(0, index)
-    }
+      )
+      rec.transientAttrs(ConsensusTags.MolecularId) = mi
+      mi
+    })
+  }
+
+  private val MolecularIdNoTrailingSuffix: String = "__" + ConsensusTags.MolecularId + "__"
```

Comment:
This could use a comment since I had to stare at it to figure out what it was!

Reply:
ctrl-f for the win?  I'll add a comment too.